app/controllers/customers_controller.rb

```diff
@@ -37,6 +37,46 @@ def create
     end
   end
 
+  def add_customer
+    if is_customer_exist(params[:customer][:phone_number])
+      @customer = Customer.find_by(phone_number: params[:customer][:phone_number])
+    else
+      @customer = Customer.new(customer_params)
+      @customer.save
+    end
+  end
+
+  def is_customer_exist(customers_phone_number)
+    Customer.find_by('phone_number = ?',customers_phone_number)
+  end
+
+  def assign_cleaner(customer_id,customers_city_id,customers_date)
+    cleaners = City.find(customers_city_id).cleaner.where(email_confirmed: true)
+    cleaners.each do |cleaner|
+      unless cleaner_available_for_date(cleaner,customers_date)
+
+        Booking.create(cleaner_id: cleaner.id,customer_id: customer_id,date: customers_date)
+
+        # Send email to cleaner for new work assignment
+        send_email(cleaner,customers_date)
+
+        return "Dear Customer, Your home cleaning duty is assign to #{cleaner.first_name}
+          #{cleaner.last_name} on the date #{customers_date}."
+      end
+    end
+    "Dear Customer, Sorry to inform you that because of heavy booking their
+      are no any cleaner available in you city for date #{customers_date}
+      you choose. Please try for another date."
+  end
+
+  def cleaner_available_for_date(cleaner,customers_date)
+    Booking.where('cleaner_id = ?',cleaner).where('date = ?',customers_date).present?
```

Comment:
@keyurgohil2426 This method should just check the count, so it should be using either `.count > 0` OR `.exists?`. Read about "How to check if data exists in database tables optimally?" - This is related to SQL, but we need to adhere the ActiveRecords way of doing it.